backend/pick-git/src/test/java/com/woowacourse/pickgit/unit/post/domain/PostRepositoryTest.java

```diff
@@ -103,25 +105,49 @@ void save_SavedPostWithCreatedDate_Success() {
         assertThat(savedPost.getCreatedAt()).isBefore(LocalDateTime.now());
     }
 
-    @DisplayName("게시글을 저장할 때 태그도 함께 영속화된다.")
+    @DisplayName("Post을 저장할 때 PostTag도 함께 영속화된다.")
     @Test
-    void save_WhenSavingPost_TagSavedTogether() {
+    void save_WhenSavingPostAndExistsTag_PostTagSavedTogether() {
+        // given
         Post post =
             new Post(null, null, new PostContent("abc"), githubRepoUrl, null, null, new ArrayList<>(),
                 null);
-        List<Tag> tags = Arrays.asList(new Tag("tag1"), new Tag("tag2"));
-        post.addTags(tags);
-        postRepository.save(post);
-        Tag entityTag = tagRepository.save(new Tag("33"));
-        post.addTags(Arrays.asList(entityTag));
+        Tag tag1 = new Tag("tag1");
+        Tag tag2 = new Tag("tag2");
+        tagRepository.save(tag1);
+        tagRepository.save(tag2);
 
         testEntityManager.flush();
         testEntityManager.clear();
 
-        Post findPost = postRepository.findAll().get(0);
+        // when
+        post.addTags(List.of(tag1, tag2));
+        Post savedPost = postRepository.save(post);
+
+        testEntityManager.flush();
+        testEntityManager.clear();
+
+        // then
+        Post findPost = postRepository.findById(savedPost.getId())
+            .orElse(null);
 
-        assertThat(findPost.getTags()).hasSize(3);
-        assertThat(tagRepository.findAll()).hasSize(3);
+        assertThat(findPost).isNotNull();
```

Comment:
```suggestion
        assertThat(findPost.getId()).isNotNull();
```

객체가 null이 아닌지 검증하는 것보다 객체 id가 null이 아닌지 검증하는 게 더 낫지 않을까요?

Reply:
@da-nyee 객체 자체가 db에 존재하는지 확인하는 로직이라 원래가 맞는덧